### PR TITLE
migrating content from e2e.sh to separate, more granular, GCB builds

### PR DESCRIPTION
**Pull request by nlopezgi**
Moving cc_image and go_image from e2e.sh to separate GCB configs.
Will continue migrating more content from e2e tests in upcoming PRs.
Most content in new sh files is just copied from e2e.sh with minor adaptations to run safely on GCB.

### CODE REVIEW

File: testing/e2e.sh [hunk 1]
```diff
@@ -246,53 +246,12 @@ EOF
   rm -f output.txt
```
Comment (xingao267) [inline testing/e2e.sh]: import the util.sh in this as well?

File: testing/e2e/cc_image.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  cd "${ROOT}"
+  EXPECT_CONTAINS "$(bazel run "$@" testdata:cc_binary_as_image)" "Hello World"
+}
+
+function test_cc_image_wrapper() {
+  cd "${ROOT}"
+  EXPECT_CONTAINS "$(bazel run "$@" testdata:cc_image_wrapper)" "Hello World"
+}
+
+if [[ $# -ne 1 ]]; then
```
Comment (xingao267) [inline testing/e2e/cc_image.sh]: Add some doc for this.

File: testing/e2e.sh (hunk 1 above)
Comment (nlopezgi) [reply testing/e2e.sh]: done

File: testing/e2e/cc_image.sh (hunk 2 above)
Comment (nlopezgi) [reply testing/e2e/cc_image.sh]: fixed

**Comment by clintharrison** [comment]
@nlopezgi is there a way to make the results public? The e2e tests were previously publicly viewable on Travis, but it seems that's no longer the case when they're run on GCB.

**Comment by nlopezgi** [comment]
hi @clintharrison ,
apologies for the issue. I did not realize the results were not public. I'll try to figure out a make results public. In the meantime, you can run these test locally - https://cloud.google.com/cloud-build/docs/build-debug-locally - (are you able to at least see which trigger failed or do you just see the trigger id?)

**Comment by clintharrison** [comment]
@nlopezgi thanks for looking into it, and that link is definitely helpful, too. Unfortunately it doesn't seem like we're able to see which trigger, the ID is the only public info :\

**Comment by nlopezgi** [comment]
can you let me know the ID of the failing test? Also do let me know if you are able to run any of these locally with the instructions in that link.

**Comment by clintharrison** [comment]
Yep, I can run them locally! I don't actually have a failing ID right now I need to see -- but generally I was just curious what the plan was.

**Pull request merged**